VenafiPS/Public/New-TppCustomField.ps1

```diff
@@ -0,0 +1,221 @@
+        [ValidateNotNullOrEmpty()]
+        [string]$Name,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Label,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Device', 'X509 Certificate')]
+        [ValidateNotNullOrEmpty()]
+        [string[]]$Class,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('1', '2', '4', '5')]
+        [ValidateNotNullOrEmpty()]
+        [string]$Type,
+
+        [Parameter()]
+        [string[]]$AllowedValue,
+
+        [Parameter()]
+        [string]$Single,
+
+        [Parameter()]
+        [string]$Help,
+
+        [Parameter()]
+        [string]$ErrorMessage,
+
+        [Parameter()]
+        [switch]$Mandatory,
+
+        [Parameter()]
+        [switch]$Policyable,
+
+        [Parameter()]
+        [string]$RegEx,
+
+        [Parameter()]
+        [switch]$RenderHidden,
+
+        [Parameter()]
+        [switch]$RenderReadOnly,
+
+        [Parameter()]
+        [switch]$DateOnly,
+
+        [Parameter()]
+        [switch]$TimeOnly,
+
+        [Parameter()]
+        [switch] $PassThru,
+
+        [Parameter()]
+        [VenafiSession] $VenafiSession = $script:VenafiSession
+    )
+
+    begin {
+        $VenafiSession.Validate() | Out-Null
+
+        $resultObj = @()
+
+        $params = @{
+            VenafiSession = $VenafiSession
+            Method     = 'Post'
+            UriLeaf    = 'Metadata/DefineItem'
+            Body       = @{
+                "Item" = @{
+                    "Name" = $Name
+                    "Classes" = @($Class)
+                    "Label" = $Label
+                    "Type" = $Type
+                    "Mandatory" = $Mandatory.ToString().ToLower()
+                    "Policyable" = $Policyable.ToString().ToLower()
+                    "RenderHidden" = $RenderHidden.ToString().ToLower()
+                    "RenderReadOnly" = $RenderReadOnly.ToString().ToLower()
+                    "DateOnly" = $DateOnly.ToString().ToLower()
+                    "TimeOnly" = $TimeOnly.ToString().ToLower()
+                }
+            }
+        }
+        if ( $PSBoundParameters.ContainsKey('Help') ) {
+            $params.Body.Item['Help'] = $Help
+        }
+        if ( $PSBoundParameters.ContainsKey('AllowedValues') ) {
+            $params.Body.Item['AllowedValues'] = @($AllowedValues)
+        }
+        if ( $PSBoundParameters.ContainsKey('Single') ) {
+            $params.Body.Item['Single'] = $Single
+        }
+        if ( $PSBoundParameters.ContainsKey('ErrorMessage') ) {
+            $params.Body.Item['ErrorMessage'] = $ErrorMessage
+        }
+        if ( $PSBoundParameters.ContainsKey('RegEx') ) {
+            $params.Body.Item['RegularExpression'] = $RegEx
+        }
+    }
+
+    end {
+        if ( $PSCmdlet.ShouldProcess($Label) ) {
+            $response = Invoke-VenafiRestMethod @params
+
+            try {
+                if (( $PassThru ) -and ( $response.Result -eq [TppMetadataResult]::Success )) {
```

Comment:
can't check these at the same time.  need to check success first, fail on error, and then passthru

Reply:
Changed this around in latest commit. Think that may work better